02_Development/Display/Tests/test_spl06_007.py

```diff
@@ -0,0 +1,355 @@
+                                1)
+        self.assertAlmostEqual(calibrator.pressure(0, 0), 0.0,
+                               msg="Fails to return 0 pressure when all "
+                               "coefficients and data equal 0.")
+        self.assertAlmostEqual(calibrator.temperature(0), 0.0,
+                               msg="Fails to return 0 temperature when "
+                               "all coefficient and data equal 0.")
+
+    def test_zero_coefficients_and_non_zero_data(self):
+        calibrator = Calibrator((0, 0, 0, 0, 0, 0, 0, 0, 0, 0),
+                                1,
+                                1)
+        self.assertAlmostEqual(calibrator.pressure(10, 10), 0.0,
+                               msg="Fails to return 0 pressure when all "
+                               "coefficients equal 0.")
+        self.assertAlmostEqual(calibrator.temperature(10), 0.0,
+                               msg="Fails to return 0 temperature when "
+                               "all coefficient equal 0.")
+
+    def test_one_coefficients_and_data(self):
+        calibrator = Calibrator((1, 1, 1, 1, 1, 1, 1, 1, 1),
+                                1,
+                                1)
+        self.assertAlmostEqual(calibrator.pressure(1, 1), 7.0,
+                               msg="Fails to return 7.0 pressure when "
+                               "all coefficients and data equal 1.0.")
+        self.assertAlmostEqual(calibrator.temperature(1), 1.5,
+                               msg="Fails to return 1.5 temperature "
+                               "when all coefficients and data equal 1.0.")
+
+    def test_one_coefficients_and_zero_data(self):
+        calibrator = Calibrator((1, 1, 1, 1, 1, 1, 1, 1, 1),
+                                1,
+                                1)
+        self.assertAlmostEqual(calibrator.pressure(0, 0), 1.0,
+                               msg="Fails to return 1.0 pressure when "
+                               "all coefficients equal 1.0 and data "
+                               "equals 0.")
+        self.assertAlmostEqual(calibrator.temperature(0), 0.5,
+                               msg="Fails to return 1.5 temperature "
+                               "when all coefficients equal 1.0 and "
+                               "data equals 0.")
+
+    def test_zero_scaling_factor_raises_exception(self):
+        with self.assertRaises(ZeroDivisionError,
+                               msg="Fails to raise a ZeroDivisionError "
+                               "when initialized with a scaling factor "
+                               "equal to 0."):
+            calibrator = Calibrator((0, 0, 0, 0, 0, 0, 0, 0, 0, 0),
+                                    0,
+                                    0)
+
+    def test_returns_standard_pressure_given_actual_data(self):
+        # calibration coefficients pulled from a sensor
+        calibrator = Calibrator((199, -249,
```

Comment:
Lots of magical numbers in this and following functions.  Maybe some extra commentary about where they come from?  Are they ex rectum, or from docs?

Reply:
Ok you must be somehow looking at an old version, because I've definitely changed that to de-magic it.